auth/middleware.go

```diff
@@ -75,6 +81,33 @@ func (k *kratosMiddleware) Session(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
+func (k *kratosMiddleware) validateAccessToken(ctx context.Context, username, password string) (*models.AccessToken, error) {
+	query := `
+SELECT
+  access_tokens.*
+FROM
+  access_tokens
+  LEFT JOIN people ON access_tokens.person_id = people.id
+WHERE
+  people.name = ?
+  AND access_tokens.expires_at > NOW()`
+
+	var acessToken models.AccessToken
+	if err := k.db.Raw(query, username).First(&acessToken).Error; err != nil {
```

Comment:
This is the other way around, we bcrypt the token and then do a lookup for the hashed value to get the person and expiry

Reply:
If the token has expired that should be included in the error message returned - "token expired on ABC"

Reply:
> we bcrypt the token and then do a lookup for the hashed value
Current std library doesn't allow creating a bcrypt hash with a custom salt. We'd need to resort to a different hashing algorithm.

Although I think what we currently have is fine (apart from the expiry clause in the SQL query).

Reply:
We'll go with argon2